ctsit/nacculator
# LBD Short Form

This change adds the new Lewy Body Disease module version 3.1 (short version) to NACCulator's processing options. 
The LBD short version is a subset of the regular LBD module with a lot of overlapping variable names, so a REDCap project / research center will have either the standard version or the short version, not both. This branch allows NACCulator to handle both cases.

This change increases NACCulator's flexibility and keeps pace with NACC's updates as it releases new research forms.


## Verification

1. Make sure to use the LBD short version test data csv (see attached data on jira issue). 
2. From the command-line, run:
`redcap2nacc -ivp -lbdsv <data_test_lbdsv.csv >data_test_lbdsv_ivp.txt`
`redcap2nacc -fvp -lbdsv <data_test_lbdsv.csv >data_test_lbdsv_fvp.txt`
3. The output should look like normal NACCulator .txt output, with each form in the LBD short version module visible on its own line in the output (B1L, etc).
4. If you pass data with errors, the output should alert you in the terminal screen- data out of range, char fields that are too long, violated blanking rules, etc.
5. You now have ivp and fvp output data that is ready to upload to NACC.

- Verify the thing does this: Creates a NACC-importable .txt file out of a REDCap-exported CSV with LBD short version data with all fields in the proper columns.
- Verify the thing does not do that: Does not try to process as anything other than LBD short version data (and the specified ivp or fvp flag), or allow fields that cause any misalignment in the columns.


## Affirmations

All of these should have a check by them. Any exception requires an explanation.

* [x] I tested data with every previous flag and ensured my changes did not introduce errors.
